code/modules/urist/modules/shipbattles/weapons.dm

```diff
@@ -282,6 +282,112 @@
+		else if(is_rigged && attached_device)
+			to_chat(user, "<span class='notice'>You carefully begin to remove [attached_device] from the warhead's internals.</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='notice'>You carefully remove [attached_device] from the warhead's internals.</span>")
+				var/obj/item/device/assembly/A = attached_device
+				A.forceMove(get_turf(user))
+				A.holder = null
+				attached_device = null
+				is_rigged = 0
+				playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+		return
+	else if(istype(I, /obj/item/device/multitool))
+		if(!riggedstate)
+			return
+		if(!safety)
+			to_chat(user, "<span class='notice'>You begin to re-engage the built-in safeties.</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='notice'>The warhead beeps softly, indicating you have re-enabled it's safeties!</span>")
+				safety = 1
+				playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+		if(safety)
+			to_chat(user, "<span class='notice'>You begin to disable the built-in safeties...</span>")
+			if(do_after(user,40))
+				to_chat(user, "<span class='danger'>The warhead beeps stridently as you disable the built-in safeties.</span>")
+				safety = 0
+				playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+	else if(istype(I, /obj/item/stack/cable_coil) && riggedstate == 2)
+		to_chat(user, "<span class='notice'>You rig a wire from the torpedo warhead's detonator circuit. You can now attach something to it to detonate it remotely.</span>")
+		riggedstate++
+		playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+	else if(istype(I, /obj/item/device/assembly))
+		if(riggedstate == 3)
+			var/obj/item/device/assembly/A = I
+			if(A.secured)
+				to_chat(user, "<span class='notice'>The [A] is secured, and you can not attach it.</span>")
+				return
+			to_chat(user, "<span class='warning'>You rig the [A] to the torpedo warhead's detonator circuit!</span>")
+			is_rigged = 1
+			if(!user.unEquip(A, src))
+				return
+			A.forceMove(src)
+			attached_device = I
+			A.holder = src
+			log_and_message_admins("[user] has rigged a torpedo IED.")
+			playsound(src.loc, 'sound/items/Wirecutter.ogg', 50, 1)
+	return
+
+/obj/item/shipweapons/torpedo_warhead/proc/process_activation() // uh oh, time to boom
+	playsound(src.loc, 'sound/machines/buttonbeep.ogg', 25, 0, 10)
+	if(safety)
+		visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+	if(!safety)
+		if(prob(15)) // Small chance for the warhead's safeties to engage briefly.
+			visible_message("<span class='danger'>[src] beeps stubbornly, refusing to detonate!</span>")
+			playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 25, 0, 10)
+			return
+		visible_message("<span class='danger'>[src] pings, begining a short countdown!</span>")
+		playsound(src.loc, 'sound/machines/ping.ogg', 25, 0, 10)
+		spawn(40) // 4 seconds to run away.
```

Comment:
Ditch the comment, just use `spawn(4 SECONDS)`
This can also be applied through the PR to other do_afters/etc.